src/librustc_middle/ty/fold.rs

```diff
@@ -142,6 +142,11 @@ pub trait TypeFoldable<'tcx>: fmt::Debug + Clone {
         self.has_type_flags(TypeFlags::HAS_RE_LATE_BOUND)
     }
 
+    /// Indicates whether this value is further_specializable.
```

Comment:
"whether this value still has parameters/placeholders/inference variables which could be replaced later, in a way that would change the results of `impl` specialization"

Reply:
Fixed